src/crystalla/ndarray.cr

```diff
@@ -0,0 +1,523 @@
+require "./ndarray/builders"
+require "./transformations/pca"
+require "./utils/random_gaussian"
+require "./ndarray/functions"
+
+module Crystalla
+
+
+  # class Dtype(T)
+
+  #   def initialize
+  #   end
+
+  # end
+
+  class Ndarray
+    include LapackHelper
+    include BlasHelper
+
+    getter values
+    getter shape
+    setter shape
+
+    @shape : Tuple(Int32, Int32)
+    @values : Array(Float64)
+
+    def initialize(values, shape = {0,0})
+      @shape = shape
+      @shape = set_shape(values) unless shape != {0,0}
+      @values = set_values(values)
+      #TODO
+      #@dtype = set_type(values)
+    end
+
+
+    def self.[](*values : Array(Number)) : Ndarray
+      Ndarray.new(values)
+    end
+
+    def set_shape(values)
+      if values.is_a?(Array(Int32)) ||values.is_a?(Array(Float64))
```

Comment:
Weird. Did you get a compiler error or was it a syntax highlighting issue? It's compiling and passing specs with the space for me. If it's a syntax highlighting issue, it's probably due to an underfeatured editor plugin. Unsurprisingly, initial Crystal support for most editors is based on Ruby grammar.